support for drag and drop file uploads

rschick opened the pull request:
Valid files:

![Screen Shot 2020-02-25 at 10 19 03 AM](https://user-images.githubusercontent.com/423474/75274816-5e8b2e00-57b8-11ea-98e7-94250286d3c8.png)

Upload status window opens:

![Screen Shot 2020-02-25 at 10 19 28 AM](https://user-images.githubusercontent.com/423474/75274830-63e87880-57b8-11ea-9a39-16d6899c19a3.png)

Invalid file:

![Screen Shot 2020-02-25 at 10 19 44 AM](https://user-images.githubusercontent.com/423474/75274845-68149600-57b8-11ea-863f-38ef1d7bfa41.png)

Note: a mix of valid and invalid files is accepted, and only the valid files are uploaded.

rschick commented on src/components/upload-status-management.js:
```diff
@@ -212,7 +213,7 @@ class UploadStatusManagement extends InternalLocalizeMixin(RtlMixin(MobxReaction
 
 	render() {
-		return html`
+		return this.uploader.statusWindowVisible ? html`
```
@qwangd2l I switched to use uploader state to control whether the window is visible or not.

rcausey commented:
Looks good. How hard would it be to limit the upload zone to the list area (excluding the sort headers)? (like Google Drive)

qwangd2l commented on src/components/upload-status-management.js:
```diff
@@ -38,6 +38,7 @@ class UploadStatusManagement extends InternalLocalizeMixin(RtlMixin(MobxReaction
 				border-radius: 0.3rem;
 				border: 1px solid var(--d2l-color-mica);
 				bottom: 1.2rem;
+				right: 1.2rem;
```
I don't think we need this since it's being done in `_resized()`. If we do need this, we need a rtl version as well.

qwangd2l commented on src/components/test/upload-status-management.test.js:
```diff
@@ -12,7 +12,6 @@ describe('upload-status-management', () => {
 	describe('accessibility', () => {
 		beforeEach(async() => {
 			element = await fixture(html`<upload-status-management id="upload-status-management"></upload-status-management>`);
-			element.show();
```
Since `statusWindowVisible` is only set to true when calling `uploadFiles` (which we don't do in this test), we'll need to manually set it to make it visible:
`element.uploader.statusWindowVisible = true`

qwangd2l commented on src/pages/content-page.js:
```diff
@@ -1,12 +1,11 @@
 import { css, html } from 'lit-element/lit-element.js';
 import { heading2Styles } from '@brightspace-ui/core/components/typography/styles.js';
+import { PageViewElement } from '../components/page-view-element.js';
+import { DependencyRequester } from '../mixins/dependency-requester-mixin.js';
```
I don't see any dependency requesting. Just leftover code maybe?

qwangd2l commented on src/state/uploader.js:
```diff
@@ -35,6 +36,7 @@ export class Uploader {
 					}
 				});
 				yield uploader.upload();
+				yield this.apiClient.processRevision({ contentId: content.id, revisionId: revision.id });
```
This is probably for another PR:
- we need to change progress to include this workflow of processing revision

rschick commented:
Updated to just highlight the list area:

![Screen Shot 2020-03-02 at 8 31 59 AM](https://user-images.githubusercontent.com/423474/75703594-9b4d9e00-5c6c-11ea-9983-8c44769e7919.png)

Removed JS resize behavior to just use CSS, but this means the popup goes into the gutter. We can revisit in future if needed:

![Screen Shot 2020-03-02 at 8 32 22 AM](https://user-images.githubusercontent.com/423474/75703678-c20bd480-5c6c-11ea-9947-67d1debf5416.png)
